engines/unknown-game-dialog.cpp

```diff
@@ -0,0 +1,144 @@
+
+	_reportData = reportData;
+	_reportTranslated = reportTranslated;
+	_bugtrackerAffectedEngine = bugtrackerAffectedEngine;
+
+	//Check if we have clipboard functionality and expand the reportTranslated message if needed...
+	if (g_system->hasFeature(OSystem::kFeatureClipboardSupport)) {
+		_reportTranslated += _("Use the button below to copy the required game information into your clipboard. \n");
+	}
+
+	//Check if we have support for opening URLs and expand the reportTranslated message if needed...
+	if (g_system->hasFeature(OSystem::kFeatureOpenUrl)) {
+		_reportTranslated += _("You can also directly report your game to the Bug Tracker!");
+	}
+
+	const int screenW = g_system->getOverlayWidth();
+	const int screenH = g_system->getOverlayHeight();
+
+	int buttonWidth = g_gui.xmlEval()->getVar("Globals.Button.Width", 0);
+	int buttonHeight = g_gui.xmlEval()->getVar("Globals.Button.Height", 0);
+
+	//Calculate the size the dialog needs
+	Common::Array<Common::String> lines;
+	int maxlineWidth = g_gui.getFont().wordWrapText(_reportTranslated, screenW - 2 * 20, lines);
+
+	_w = MAX(MAX(maxlineWidth, 0), (2 * buttonWidth) + 10) + 20;
+	int lineCount = lines.size() + 1;
+
+	_h =  3 * kLineHeight;
+	_h += lineCount * kLineHeight;
+
+	// Center the dialog
+	_x = (screenW - _w) / 2;
+	_y = (screenH - _h) / 2;
+
+	// Buttons
+	int closeButtonWidth = MAX(buttonWidth, g_gui.getFont().getStringWidth(_("Close")) + 10);
+	int copyToClipboardButtonWidth = MAX(buttonWidth, g_gui.getFont().getStringWidth(_("Copy to clipboard")) + 10);
+	int openBugtrackerURLButtonWidth = MAX(buttonWidth, g_gui.getFont().getStringWidth(_("Report game")) + 10);
+	int closeButtonPos = _w - closeButtonWidth - 10;
+	int copyToClipboardButtonPos = _w - copyToClipboardButtonWidth - closeButtonWidth - 15;
+	int openBugtrackerURLButtonPos = _w - copyToClipboardButtonWidth - closeButtonWidth - openBugtrackerURLButtonWidth - 20;
+	_closeButton = new ButtonWidget(this, closeButtonPos, _h - buttonHeight - 8, buttonWidth, buttonHeight, _("Close"), 0, kClose);
+
+	//Check if we have clipboard functionality
+	if (g_system->hasFeature(OSystem::kFeatureClipboardSupport)) {
+		new ButtonWidget(this, copyToClipboardButtonPos, _h - buttonHeight - 8, copyToClipboardButtonWidth, buttonHeight, _("Copy to clipboard"), 0, kCopyToClipboard);
+	} else {
+			//Set the bug tracker position to the one of the disabled copy-to-clipboard button
+			openBugtrackerURLButtonPos = copyToClipboardButtonPos;
+	}
+
+	//Check if we have support for opening URLs
+	if (g_system->hasFeature(OSystem::kFeatureOpenUrl)) {
+		new ButtonWidget(this, openBugtrackerURLButtonPos, _h - buttonHeight - 8, openBugtrackerURLButtonWidth, buttonHeight, _("Report game"), 0, kOpenBugtrackerURL);
+		//Formatting the reportData for bugtracker submission [replace line breaks]...
+		_bugtrackerGameData = _reportData;
+		while (_bugtrackerGameData.contains("\n")) {
+			replace(_bugtrackerGameData, "\n", "%0A");

```

Comment:
I am confused as I would not expect this to compile (unless there are `using namespace Common;` directives scattered around). Shouldn't this line be:
```
	Common::replace(_bugtrackerGameData, "\n", "%0A");
```

Reply:
It compiled, but I changed this to your convenience ;)